### PR TITLE
Visibility analysis

### PR DESCRIPTION
**Pull request by o-**

### CODE REVIEW

**Comment by o-** [comment]
@JanJecmen again, only last commit. I'll give you this one too, since it is so similar to the redundant env elision

File: rir/src/compiler/analysis/visibility.h [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+class CurrentVisibility {
+  public:
+    enum State { Visible, Invisible, Unknown };
+
+    State state = State::Unknown;
+
+    CurrentVisibility() {}
+    CurrentVisibility(const CurrentVisibility&) = default;
+    CurrentVisibility& operator=(const CurrentVisibility&) = default;
```
Comment (JanJecmen) [inline rir/src/compiler/analysis/visibility.h]: I think you don't need to type any of these explicitly

File: rir/src/compiler/analysis/visibility.h [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+#include "generic_static_analysis.h"
+
+namespace rir {
+namespace pir {
+
+class CurrentVisibility {
+  public:
+    enum State { Visible, Invisible, Unknown };
+
+    State state = State::Unknown;
```
Comment (JanJecmen) [inline rir/src/compiler/analysis/visibility.h]: Just `Unknown` to be consistent?

**Review by JanJecmen — APPROVED** [review]
:+1:

**Pull request merged**